### PR TITLE
CLN16668: remove OrderedDefaultDict

### LINKED ISSUE #16668: CLN: private impl of OrderedDefaultDict can be removed
https://github.com/pandas-dev/pandas/blob/master/pandas/compat/__init__.py#L376

I think this was leftover from 2.6 compat.

### PR DESCRIPTION
**Pull request by faic**
- [x] closes #16668
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

The replacement of usage covered by tests.

### CODE REVIEW

File: pandas/core/panel.py [hunk 1]
```diff
@@ -262,9 +262,11 @@ def from_dict(cls, data, intersect=False, orient='items', dtype=None):
         """
         orient = orient.lower()
         if orient == 'minor':
-            new_data = OrderedDefaultdict(dict)
+            new_data = OrderedDict()
```
Comment (TomAugspurger) [inline pandas/core/panel.py]: You should be able to replace this with `new_data = defaultdict(OrderedDict)`

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -129,7 +129,7 @@ Removal of prior version deprecations/changes
 - ``Categorical`` has dropped the ``.order()`` and ``.sort()`` methods in favor of ``.sort_values()`` (:issue:`12882`)
 - :func:`eval` and :method:`DataFrame.eval` have changed the default of ``inplace`` from ``None`` to ``False`` (:issue:`11149`)
 - The function ``get_offset_name`` has been dropped in favor of the ``.freqstr`` attribute for an offset (:issue:`11834`)
-
+- The class ``OrderedDefaultDict`` has been removed in favor of ``OrderedDict`` (:issue:`16668`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.21.0.txt]: Probably can just remove this line.  This wasn't part of the documented API.

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**